tests/io/test_nexrad_level2.py

```diff
@@ -0,0 +1,353 @@
+    # #         else:
+    # #             print("--->:", name, bh)
+    # #
+    # # import matplotlib.pyplot as plt
+    # #
+    # # plt.figure()
+    # # print(fh.data[0]["sweep_data"]["PHI"]["data"])
+    # # plt.imshow(fh.data[0]["sweep_data"]["REF"]["data"])
+    # # plt.figure()
+    # # plt.imshow(fh.data[0]["sweep_data"]["ZDR"]["data"])
+    # # plt.figure()
+    # # plt.imshow(fh.data[0]["sweep_data"]["PHI"]["data"])
+    # # plt.figure()
+    # # plt.imshow(fh.data[0]["sweep_data"]["RHO"]["data"])
+    # # plt.show()
+    #
+    # # print(fh._data[0]["sweep_data"]["REF"])
+    #
+    # # for sidx in scans_idx:
+    # #     msg = fh._msg_31_headers[sidx]
+    # #     block_pointers = [
+    # #         v for k, v in msg.items() if k.startswith("block_pointer") and v > 0
+    # #     ]
+    # #     print(block_pointers)
+    # #     print(msg)
+    # #     # for block_pointer in block_pointers[:msg["block_count"]]:
+    # #     #     fh.rh.pos = block_pointer + LEN_MSG_HEADER
+    # #     #     # print(block_pointer)
+    # #     #     # print(self.filepos)
+    # #     #
+    # #     #     dheader = _unpack_dictionary(self._rh.read(4, width=1), DATA_BLOCK_HEADER,
+    # #     #                                  self._rawdata, byte_order=">")
+    # #     #
+    # #     #     block = DATA_BLOCK_TYPE_IDENTIFIER[dheader["block_type"]][
+    # #     #         dheader["data_name"]]
+    # #     #     LEN_BLOCK = struct.calcsize(_get_fmt_string(block, byte_order=">"))
+    # #     #     block_header = _unpack_dictionary(
+    # #     #         self._rh.read(LEN_BLOCK, width=1),
+    # #     #         block,
+    # #     #         self._rawdata,
+    # #     #         byte_order=">",
+    # #     #     )
+    # #
+    # # nscans = len(scan_msgs)
+    # #
+    # # print("nscans:", nscans)
+    #
+    # # print(fh._msg_5_data["elevation_data"])
+    # # print([(rec["type"], rec["size"], rec["seg_num"]) for rec in fh.raw_product_bhdrs])
+    #
+    # # elev_nums = np.array(
+    # #     [m["elevation_number"] for m in fh.raw_product_bhdrs]
+    # # )
+    # # print(elev_nums)
+    # # msg_18 = xradar.io.backends.nexrad_level2_new.MSG_18
+    # # print(xradar.io.backends.nexrad_level2_new.LEN_MSG_18)
+    # # print(xradar.io.backends.iris._get_fmt_string(msg_18, byte_order=">"))
+    # # for k, v in msg_18.items():
+    # #    print(k, v)
+    # # import time
+    # # time.sleep(3)
+    # # assert 1 == 2
+
+
+def test_open_nexrad_level2_file_compressed():
+    fname_bz = "/home/kai/data/daten/daten/radar_all_over_the_world/NEXRAD/KATX20160601_090111_V06_bz2"
+    fh = NEXRADLevel2File(fname_bz)
+    print("tell", fh._fp.tell())
+    for k, v in fh.meta_header.items():
+        print(k, v)
+
+    print("MSG_5", fh.msg_5)
+
+    print(len(fh.data_header))
+
+
+def test_open_nexrad_level2_file_uncompressed():
+    fname_gz = "/home/kai/data/daten/daten/radar_all_over_the_world/NEXRAD/KATX20160601_090111_V06_gz"
```

Comment:
@kmuehlbauer - we should have one of these sample files over in open-radar-data?

Reply:
@mgrover1 Yes, the nice thing is, we can have one with the old uncompressed data and the same file with the bzip2 compressed records. I'll add this next week.

And sorry for breaking the PR here, I'm currently testing with the compression. Seems like I'll need to change the class inheritance, but overall it seems to work.

Reply:
done